base/filesystem.jl

```diff
@@ -104,7 +104,7 @@ function sendfile(dst::File, src::File, src_offset::Int64, bytes::Int)
     nothing
 end
 
-function write(f::File, buf::Ptr{UInt8}, len::Integer, offset::Integer=-1)
+function unsafe_write(f::File, buf::Ptr{UInt8}, len::UInt, offset::Int64=Int64(-1))
```

Comment:
i also renamed it.....

i'll attempt to put in a deprecations for this. the problem is that `write(io, p, l)` is supposed to be the same as `write(io, p); write(io, l)`, but many IO streams were abusing some variant of this method signature to express the `unsafe_write` concept.